src/erliam_srv.erl

```diff
@@ -80,55 +75,109 @@ terminate(_Reason, _State) ->
 code_change(_OldVsn, State, _Extra) ->
     {ok, State}.
 
-
 %%%% INTERNAL FUNCTIONS
 
 maybe_update_credentials() ->
     case ets:lookup(?TAB, credentials) of
-        [Credentials] ->
-            MinLifetime = ?MIN_LIFETIME,
-            case remaining_lifetime(Credentials) of
-                N when N =< MinLifetime ->
-                    update_credentials();
-                _ ->
-                    ok
-            end;
-        [] ->
-            update_credentials()
+      [Credentials] ->
+          MinLifetime = ?MIN_LIFETIME,
+          case remaining_lifetime(Credentials) of
+            N when N =< MinLifetime ->
+                update_credentials();
+            _ ->
+                ok
+          end;
+      [] ->
+          update_credentials()
     end.
 
-
 update_credentials() ->
     case erliam:get_session_token() of
-        #credentials{} = Credentials ->
-            ets:insert(?TAB, Credentials),
-            ok;
-        Error ->
-            error_logger:error_msg("failed to obtain session token: ~p", [Error]),
-            Error
+      #credentials{} = Credentials ->
+          ets:insert(?TAB, Credentials),
+          ok;
+      Error ->
+          error_logger:error_msg("failed to obtain session token: ~p", [Error]),
+          Error
     end.
 
-
 remaining_lifetime(#credentials{expiration = ExpTime}) ->
-    max(0, calendar:datetime_to_gregorian_seconds(parse_exptime(ExpTime)) -
-            calendar:datetime_to_gregorian_seconds(calendar:universal_time())).
-
-
-parse_exptime([Y1,Y2,Y3,Y4, $-, Mon1,Mon2, $-, D1,D2, $T, H1,H2, $:, Min1,Min2, $:, S1,S2, $Z]) ->
-    {{list_to_integer([Y1,Y2,Y3,Y4]),
-      list_to_integer([Mon1,Mon2]),
-      list_to_integer([D1,D2])},
-     {list_to_integer([H1,H2]),
-      list_to_integer([Min1,Min2]),
-      list_to_integer([S1,S2])}};
-parse_exptime(
-    [Y1,Y2,Y3,Y4, $-, Mon1,Mon2, $-, D1,D2, $T, H1,H2, $:, Min1,Min2, $:, S1,S2, $.,_,_,_, $Z]) ->
-    parse_exptime([Y1,Y2,Y3,Y4, $-, Mon1,Mon2, $-, D1,D2, $T, H1,H2, $:, Min1,Min2, $:, S1,S2, $Z]).
-
-
+    max(0,
+        calendar:datetime_to_gregorian_seconds(parse_exptime(ExpTime)) -
+          calendar:datetime_to_gregorian_seconds(calendar:universal_time())).
+
+parse_exptime([Y1,
```

Comment:
Controversial comment, I liked more the old version

Reply:
I can fix that with some configuration… Hold on :)

Reply:
Fixed